services/continuous_querier/service.go

```diff
@@ -167,8 +167,10 @@ func (s *Service) Run(database, name string, t time.Time) error {
 		// Loop through CQs in each DB executing the ones that match name.
 		for _, cq := range db.ContinuousQueries {
 			if name == "" || cq.Name == name {
-				// Reset the last run time for the CQ.
-				s.lastRuns[cq.Name] = time.Time{}
+				// Remove the last run time for the CQ
+				if _, ok := s.lastRuns[cq.Name]; ok {
+					delete(s.lastRuns, cq.Name)
```

Comment:
The initial run of a continuous query acts differently than one that has previous been run. One that has previously been run will try to calculate the optimal time it _should_ have run the next query (so as to calculate the oldest interval to regenerate). The initial run ignores this.

This line of code set the time to January 1, 1970 and caused the code attempting to calculate the intervals to try running every query from 1970. Instead of trying to differentiate a default value vs a real value, I just decided removing it from the map was a better solution.